### PR TITLE
Refactor the whole test page

### PR DESCRIPTION
**Pull request by robertsipka**
Supporting malloc, jerry-heap, and stack information.

### CODE REVIEW

File: js/chart_settings.js [hunk 1]
```diff
@@ -159,11 +159,14 @@ function update_chart(from, to) {
 
       data.average_memory = 0;
       data.tests.forEach(function(testname) {
-        var number = parseInt(Number(testname.memory));
-        if (number == testname.memory) {
-          memory_counter++;
-          data.average_memory += number;
+        if (testname.hasOwnProperty('memory')){
```
Comment (LaszloLango) [inline js/chart_settings.js]: missing space before `{`

File: js/render_entries.js [hunk 2]
```diff
@@ -154,31 +166,54 @@ function _render_tests_table(uid, tests, earlier_tests_map) {
...
+        raw_html += '<td class="mono">' + test.memory.malloc + ' B</td>';
+      else
+        raw_html += '<td class="mono">' + test.memory.malloc + '</td>';
+
+
+      if (test.memory.stack != 'n/a')
+        raw_html += '<td class="mono">' + test.memory.stack + ' B</td>';
+      else
+        raw_html += '<td class="mono">' + test.memory.stack + '</td>';
+    }
```
Comment (LaszloLango) [inline js/render_entries.js]: inconsistent style, `} else {`

**Comment by rtakacs** [comment]
LGTM

File: js/render_entries.js [hunk 3]
```diff
@@ -154,31 +166,54 @@ function _render_tests_table(uid, tests, earlier_tests_map) {
...
 
     if (can_compare_current) {
+      var compared = false;
       for (var key in earlier_tests_map) {
         var earlier_test = earlier_tests_map[key];
-        if (earlier_test.name == test.name && earlier_test.result == "pass" && !isNaN(parseInt(earlier_test.memory))) {
-          var diff = test.memory - earlier_test.memory;
-          var percent = (diff / earlier_test.memory) * 100.0;
+        if (earlier_test.name == test.name && earlier_test.result == "pass" &&
+            earlier_test.hasOwnProperty("memory") && !isNaN(parseInt(earlier_test.memory.total)))
```
Comment (LaszloLango) [inline js/render_entries.js]:
```js
if (earlier_test.name == test.name
    && earlier_test.result == "pass"
    && earlier_test.hasOwnProperty("memory")
    && !isNaN(parseInt(earlier_test.memory.total))) {
```

**Comment by robertsipka** [comment]
@rtakacs  @LaszloLango : Thanks for the reviews, I've updated it accordingly.

**Pull request merged**